src/map/utils/trustutils.cpp

```diff
@@ -344,18 +349,45 @@ namespace trustutils
         auto adjustedDamage   = baseDamage * damageMultiplier;
         auto finalDamage      = std::max(adjustedDamage, 1.0f);
 
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_MAIN]))->setDamage(static_cast<uint16>(finalDamage));
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_RANGED]))->setDamage(static_cast<uint16>(finalDamage));
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_AMMO]))->setDamage(static_cast<uint16>(finalDamage));
+        // Trust do not really have weapons, but they are modelled internally as
+        // if they do.
+        if (auto* mainWeapon = dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_MAIN]))
+        {
+            mainWeapon->setMaxHit(1);
+            mainWeapon->setSkillType(trustData->cmbSkill);
+
+            mainWeapon->setDamage(static_cast<uint16>(finalDamage));
+            mainWeapon->setDelay((trustData->cmbDelay * 1000) / 60);
+            mainWeapon->setBaseDelay((trustData->cmbDelay * 1000) / 60);
+        }
 
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_MAIN]))->setDelay((trustData->cmbDelay * 1000) / 60);
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_MAIN]))->setBaseDelay((trustData->cmbDelay * 1000) / 60);
+        if (auto* subWeapon = dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_SUB]))
+        {
+            subWeapon->setDamage(static_cast<uint16>(finalDamage));
+            subWeapon->setDelay((trustData->cmbDelay * 1000) / 60);
+            subWeapon->setBaseDelay((trustData->cmbDelay * 1000) / 60);
+        }
+
+        if (auto* rangedWeapon = dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_RANGED]))
+        {
+            rangedWeapon->setDamage(static_cast<uint16>(finalDamage));
+            rangedWeapon->setDelay((trustData->cmbDelay * 1000) / 60);
+            rangedWeapon->setBaseDelay((trustData->cmbDelay * 1000) / 60);
+        }
 
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_RANGED]))->setDelay((trustData->cmbDelay * 1000) / 60);
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_RANGED]))->setBaseDelay((trustData->cmbDelay * 1000) / 60);
+        if (auto* ammoWeapon = dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_AMMO]))
+        {
+            ammoWeapon->setDamage(static_cast<uint16>(finalDamage));
+            ammoWeapon->setDelay((trustData->cmbDelay * 1000) / 60);
+            ammoWeapon->setBaseDelay((trustData->cmbDelay * 1000) / 60);
+        }
 
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_AMMO]))->setDelay((trustData->cmbDelay * 1000) / 60);
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_AMMO]))->setBaseDelay((trustData->cmbDelay * 1000) / 60);
+        if (trustData->mJob == JOB_NIN ||
```

Comment:
```cpp
if (trust->cmbSkill == 2)
{
    PTrust->m_dualWield = true;
}
```
I think thats the value we use on mobs pools anyway..

Reply:
```
    SKILL_NONE         = 0,
    SKILL_HAND_TO_HAND = 1,
    SKILL_DAGGER       = 2,
    SKILL_SWORD        = 3,
    SKILL_GREAT_SWORD  = 4,
    SKILL_AXE          = 5,
    SKILL_GREAT_AXE    = 6,
    SKILL_SCYTHE       = 7,
    SKILL_POLEARM      = 8,
    SKILL_KATANA       = 9,
    SKILL_GREAT_KATANA = 10,
    SKILL_CLUB         = 11,
    SKILL_STAFF        = 12,
    ```

We set AAHM to 3 for  Swords, I wonder if we need some other way of handling this, since `MOBMOD_DUAL_WIELD` is marked as "please deprecate me"

Reply:
yeah this is why calling it combat skill and pretending these object HAVE an actual weapon is bad. What we need in the future are to rip that column out and replace it with just 2 values: what damage type it is, and is it double hit (for dw and h2h). Mimas in upper delkfut does not have a great axe, his model looks like it does. AAHM does not have swords, his model looks like it does. All mobs basically are h2h behind the scenes even if they single hit, and some have their attacks count as various dmg types (can use bst pets to confirm their damage type) and the same goes for trusts. Shantotto II's melee counts as magic damage (she can damage through pld's invincible) so the damage type and "skill" are not even connected for them like players. AAHM would be doublehit with slashing damage on retail, not "sword". For our purposes, we set it to 1 so he will double hit and disregard my previous mistake of "2". All values here other than 1 essentially do the same thing for us.

Check all our existing ninja mobs that dw. you will see all their pools say 1.

Reply:
edited for readability/clearity.